roaring64/bsi64.go

```diff
@@ -378,19 +475,35 @@ func (b *BSI) UnmarshalBinary(bitData [][]byte) error {
 			continue
 		}
 		if b.BitCount() < i {
-			b.bA = append(b.bA, NewBitmap())
+			newBm := NewBitmap()
+			if b.runOptimized {
+				newBm.RunOptimize()
+			}
+			b.bA = append(b.bA, newBm)
 		}
 		if err := b.bA[i-1].UnmarshalBinary(bitData[i]); err != nil {
 			return err
+		} else {
```

Comment:
The else is unnecessary and the rest of the code should not be in braces.

Reply:
Done.